api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -27,16 +38,125 @@ public NameSupport() {
 	}
 	
 	public static NameSupport getInstance() {
-		if (singleton == null) {
-			throw new APIException("Not Yet Instantiated");
-		} else {
```

Comment:
I would not remove this logic as it helpful in some instances. To make the changes backwards compatible, I'd change the logic to look something like;

```
public static NameSupport getInstance() {
        NameSupport customizedNameSupport = init();
        if (customizedNameSupport != null) {
            singleton = customizedNameSupport;
        }
		if (singleton == null) {
			throw new APIException("Not Yet Instantiated");
		} else {
             return singleton;
        }
}
```

This implies `init()` returns `null` if `GLOBAL_PROPERTY_LAYOUT_NAME_TEMPLATE` wasn't configured.